### DISCUSSION TITLE
Recovery from backup

### DISCUSSION THREAD

**Discussion by adamryczkowski** [Kozea/Radicale#1391] [Q&A general]
Hi! 

The radicale installation I have went missing (however improbable it sounds ;-) ). Fortunately I have a backup of the whole container hosting the instance. Unfortunately, it is many months old. Fortunately I have four Android devices synchronized to that instance, which potentially can be used to recover the data. 

When I re-set the radicale server back, it will be in a time-shifted state, potentially confusing the clients. What should I do to transfer the calendar data from a client(s) to the restored server? Would simple CalDav synchronization suffice?

**Comment by zootboy** [comment]
I just encountered a similar situation myself, and I did some tests to see how a Radicale + DAVx<sup>5</sup> setup would handle having the Radicale server be spontaneously rolled back. The answer is: very, very poorly.

**Do not roll back a Radicale server, you WILL lose data!**

In my experiments with contacts, calendar events (DAVx<sup>5</sup>), and tasks (Tasks.org) sync, if you roll back the Radicale server, every event, contact, and task that has been modified on-phone between the rollback point and now will be _deleted_.

Now, this isn't a slight against Radicale; I'm not certain which element (Radicale, DAVx, or even the *DAV protocol in general) is at fault here. I'm still trying to figure out a path forward for my own situation. I'm hoping to stand up a new, fresh Radicale server and get my devices to sync with it. I'll be taking plenty of on-device backups beforehand just in case things go sideways.

EDIT: After doing some more reading on the internal architecture of Radicale and the DAVx docs, I'm mostly convinced that the issue is on DAVx's end. My migration plan may be harder than anticipated; see this: https://www.davx5.com/faq/synchronization/change-base-url for a doozy of a disclaimer:

> Attention: In the case that DAVx⁵ can't synchronize because your old URL isn't accessible anymore, there may be unsynchronized local changes (for instance, a new event). These changes will be deleted when you delete the old account.
> 
> Currently there is no good solution for that problem.

It's looking like I'll need to do a careful manual export and import of my contacts and calendars from a single known-good device. I don't think there's any easy way of auto-merging multiple devices that have diverging calendars or contact books.
